python/setup.py

```diff
@@ -301,6 +303,78 @@ def download_pickle5(pickle5_dir):
                 wzf.close()
 
 
+def patch_isdir():
+    """
+    Python on Windows is having hard times at telling if a symlink is
+    a directory - it can "guess" wrong at times, which bites when
+    finding packages. Replace with a fixed version which unwraps links first.
+    """
+    orig_isdir = os.path.isdir
+
+    def fixed_isdir(path):
+        while os.path.islink(path):
+            try:
+                link = os.readlink(path)
+            except OSError:
+                break
+            path = os.path.abspath(os.path.join(os.path.dirname(path), link))
+        return orig_isdir(path)
+
+    os.path.isdir = fixed_isdir
+
+
+def replace_symlinks_with_junctions():
```

Comment:
Make a comment about when you need a junction vs when you need a symlink?

i.e., what is the purpose of this function and why do we need it?

Reply:
cc @gshimansky @vnlitvinov can you help me here? There's some information online but maybe you can help why this was introduced initially

Reply:
The thing is, default settings on Windows require admin rights to make symlinks (for whatever reason MS decided to do so) while junctions work with normal accounts (but for directories only).

That said, I think Azure CI runs with admin rights, but on users' systems it could be not so, and checking out a repo with symlinks on Windows could lead to _very_ peculiar state where symlinks would actually be text files (like they're on POSIX but without special attribute to interpret them differently).

So this function tries to "repair" a repository so it works for both admin and non-admin accounts to be buildable.
Maybe it should be applied under special variable like for `CONDA_BUILD`, too...

Reply:
Thanks! I'll add this as a comment to the function. It is only called if `is_automated_build==True`, so that shouldn't be a problem in regular contexts.